trie.go

```
@@ -17,8 +17,61 @@ type Trie struct {
+	} else {
+		b.arena = b.arena[:n]
+	}
+	if cap(b.ptrs) < n {
+		b.ptrs = make([]*Match, n)
+	} else {
+		b.ptrs = b.ptrs[:n]
+	}
+	for k := 0; k < n; k++ {
+		end := b.raw[2*k]
+		dp := b.raw[2*k+1]
+		ln := uint32(dp)
```

golangci-lint (2.12.2)

[high] G115: integer overflow conversion uint64 -> uint32

(gosec)